acquia/cli
# CLI-692: Remove webmozart.

**Motivation**

Enables https://github.com/acquia/cli/issues/782

**Proposed changes**

* Remove all implementations of webmozart/key-value-store
* Add our own datastore classes for `cloud_api.conf` and `.acquia-cli.yml`
* Add validation to datastore classes

There is a notable internal change here. Prior to this, values from the datastore would not be loaded until requested. We were not really injecting configuration into any services, we were injecting a datastore and lazy loading values. 

Now, we load the configuration up front and inject it as a dependency into other services. 

This change required various changes to testing so that configuration is set and loaded earlier.

**Alternatives considered**

* Upgrading webmozart ourselves

**Testing steps**


1. Follow the [contribution guide](https://github.com/acquia/cli/blob/master/CONTRIBUTING.md#building-and-testing) to set up your development environment.
2. Clear the kernel cache to pick up new and changed commands: `./bin/acli ckc`
3. Nothing should appear to have changed!
4. Run `acli login`
5. Run `acli link`
6. Run `acli logout`

**Merge requirements**
